tools/ResourceMetaSchema.json

```diff
@@ -1,108 +1,200 @@
+    },
+    "resource": {
+      "type": "object",
+      "properties": {
+        "type": { "enum": [ "object" ] },
+        "description": { "$ref": "#/definitions/description" },
+        "required": { "$ref": "#/definitions/stringArrayEmptyOkay" },
+        "properties": {
+          "type": "object",
+          "minProperties": 1,
+          "properties": {
+            "type": { "$ref": "#/definitions/resourceTypeProperty" },
+            "apiVersion": { "$ref": "#/definitions/apiVersionProperty" },
+            "resources": { "$ref": "#/definitions/childResources" },
+            "properties": { "$ref": "#/definitions/resourceProperties" }
+          },
+          "required": [ "apiVersion", "type" ]
+        }
+      },
+      "required": [ "description" ]
+    },
+    "resourceTypeProperty": {
+      "type": "object",
+      "properties": {
+        "enum": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "minLength": 1
+          },
+          "minItems": 1
+        }
+      },
+      "required": [ "enum" ],
+      "additionalProperties": false
+    },
+    "resourceProperties": {
+      "type": "object",
+      "properties": {
+        "type": { "enum": [ "object" ] },
+        "$ref": {
+          "type": "string",
+          "format": "uri"
         },
-        "resourceProperty": {
-            "allOf": [
-                { "$ref": "http://json-schema.org/draft-04/schema#" },
-                {
-                    "type": "object",
-                    "properties": {
-                        "description": { "$ref": "#/definitions/description" }
-                    },
-                    "required": [ "description" ],
-                    "additionalProperties": true
-                }
-            ]
+        "properties": {
+          "type": "object",
+          "additionalProperties": { "$ref": "#/definitions/resourceProperty" }
         },
-        "childResources": {
-            "type": "object",
-            "required": [ "type", "items", "description" ],
-            "properties": {
-                "type": { "enum": [ "array" ] },
-                "description": { "$ref": "#/definitions/description" },
-                "items": { "type": "object" }
-            }
+        "required": { "$ref": "#/definitions/stringArrayEmptyOkay" },
+        "additionalProperties": { "$ref": "http://json-schema.org/draft-04/schema#/properties/additionalProperties" }
+      },
+      "additionalProperties": false
+    },
+    "resourceProperty": {
+      "type": "object",
+      "properties": {
+        "$ref": {
+          "type": "string",
+          "format": "uri"
         },
-        "description": {
-            "type": "string",
-            "minLength": 1
+        "items": { "$ref": "http://json-schema.org/draft-04/schema#/properties/items" },
+        "oneOf": { "$ref": "#/definitions/resourcePropertyDefinitionOptions" },
+        "type": { "enum": [ "string" ] },
```

Comment:
this seems not correct. property under "properties" of a resource should be a object.